### PR TITLE
Rely on numel() == 1 to check if distribution parameters are scalar.

### PR DESCRIPTION
**Pull request by mfuntowicz**
As discussed here #16952, this PR aims at improving the __repr__ for distribution when the provided parameters are torch.Tensor with only one element.

Currently, __repr__() relies on dim() == 0 leading to the following behaviour :

```
>>> torch.distributions.Normal(torch.tensor([1.0]), torch.tensor([0.1]))
Normal(loc: torch.Size([1]), scale: torch.Size([1]))
```

With this PR, the output looks like the following:
```
>>> torch.distributions.Normal(torch.tensor([1.0]), torch.tensor([0.1]))
Normal(loc: 1.0, scale: 0.10000000149011612)
```

### CODE REVIEW

File: torch/distributions/distribution.py [hunk 1]
```diff
@@ -260,7 +260,7 @@ def _get_checked_instance(self, cls, _instance=None):
 
     def __repr__(self):
         param_names = [k for k, _ in self.arg_constraints.items() if k in self.__dict__]
-        args_string = ', '.join(['{}: {}'.format(p, self.__dict__[p]
-                                if self.__dict__[p].dim() == 0
+        args_string = ', '.join(['{}: {}'.format(p, self.__dict__[p].item()
```
Comment (vishwakftw) [inline torch/distributions/distribution.py]: I'm ambivalent about the requirement of `.item()` here.

Comment (ezyang) [reply torch/distributions/distribution.py]: I agree. I went ahead and axed it.

**Comment by vishwakftw** [comment]
@ezyang I believe test failures are spurious. Otherwise, the PR looks good to me.

**Review by soumith — APPROVED** [review]
thanks!
